spec/mongo/bulk_write_spec.rb

```diff
@@ -8,6 +8,14 @@
 
   after do
     authorized_collection.delete_many
+    collection_with_validator.drop()
+  end
+
+  let(:collection_with_validator) do
+    c = authorized_client[:validating,
+                          :validator => {:a => {'$exists' => true} }]
+    c.create
+    c
```

Comment:
ok, that makes sense because in order for the collection options to be applied, you actually have to call the create collection command. Otherwise, this let block will return the collection object and just start using it, without calling create to apply the options.

In that case, I would write the block as:

```ruby
let(:collection_with_validator) do
  authorized_client[:validating, :validator => {:a => {'$exists' => true} }].tap do |c|
    c.create
  end
end
```

#tap runs a block on an object and returns the object when the block is done. It's more Ruby-like = )